pygmt/clib/conversion.py

```diff
@@ -95,9 +97,12 @@ def dataarray_to_matrix(grid):
         coord_incs = coord[1:] - coord[0:-1]
         coord_inc = coord_incs[0]
         if not np.allclose(coord_incs, coord_inc):
```

Comment:
Could we try to work out the [`np.assert_allclose`](https://numpy.org/doc/1.21/reference/generated/numpy.allclose.html) atol or rtol argument values which won't matter for GMT? I remember reading somewhere that GMT uses [single-precision (4-byte/32-bit) floating point](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) to store the grids (see https://github.com/GenericMappingTools/gmt/blob/32296e512b173cfdd2c3e8da1d2735e49b79f02a/doc/rst/source/explain_float.rst_#L5-L7). So how many decimal places is that?

Reply:
Although GMT grids are stored using 4-bytes, the grid metadata (range, inc, and others) are stored using double-precision floating points (8 bytes) (see https://github.com/GenericMappingTools/gmt/blob/8ee9060d8620a7d0e599e3f4990303b1517aed02/src/gmt_resources.h#L413).

I'm looking at @leouieda's example data in #1468. The "longitude" dimension has the following values:
```python
>>> grid.coords["longitude"].values
array([-180.    , -179.8333, -179.6667, ...,  179.6667,  179.8333,
        180.    ], dtype=float32)
```
As you can see, the grid spacings can be either 0.1666 or 0.1667 even if we ignore the floating-point precision issue.